### PR TITLE
Translated some docs pages.

### PR DESCRIPTION
**Pull request by hvijaycse**
Translated some of the documents. 

This time I didn't copy-paste to google translator, I used AWS DL translator for more accurate result and only translated meaningful paragraph and I also make sure that any of the code doesn't get affected due to translation

What remaining in those translated page are

- Comments [ will do it next time]
- Links are not updated for Spanish language webpages

### CODE REVIEW

**Comment by Darking360** [comment]
@horacioh Just taking a look at these ones, but not quite sure if this is what we want, even though some words make sense, a good amount of meanings are lost with this translation, it's ok to have help from translation engines @hvijaycse but we want to keep the core meaning of the documentation, and we kind of lost it here. 

@horacioh Can you take a quick look at this? We can agree on what to do later 🕐

**Comment by hvijaycse** [comment]
@Darking360 can you help me to understand what's wrong with these translation, so that next time i won't repeat that .

Thank you

File: docs/docs/add-404-page.md [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-errors.
+Gatsby asegura que su página 404 se construya como `404.html` tantos hosting estático
+de forma predeterminada para usar esto como su página de error 404. Si estás alojando tu
+de otra manera, deberá configurar una regla personalizada para servir este archivo para 404
+errores.
 
-Because Gatsby creates this page for you by default, there is no need to configure
-it in your `gatsby-node.js` file.
+Como Gatsby crea esta página de forma predeterminada, no es necesario configurar
+en su `gatsby-node.js` file.
```
Comment (Darking360) [inline docs/docs/add-404-page.md]: For example here, the translation is weird, the idea would be for line 14 to end with "configurarlo" and then line 15 would be "en tu archivo `gatsby-node.js`".

File: docs/docs/adding-search-with-elasticlunr.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+title: Agregar búsqueda con elasticlunr
 ---
 
-This is a stub. Help our community expand it.
+Esto es un trozo. Ayuda a nuestra comunidad a expandirlo.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+Utilice la [Guía de estilo de Gatsby](/contributing/gatsby-style-guide/) para asegurarse de que su
+se acepta la solicitud de extracción.
```
Comment (Darking360) [inline docs/docs/adding-search-with-elasticlunr.md]: Same here, for example, the start of line 8 is weird because does not follow the end of line 7.

**Pull request closed without merging**